broker-core/src/test/java/io/zeebe/broker/workflow/TimerStartEventTest.java

```diff
@@ -117,28 +119,39 @@ public void shouldCreateTimer() {
   public void shouldTriggerAndCreateWorkflowInstance() {
     // when
     final ExecuteCommandResponse response = testClient.deployWithResponse(SIMPLE_MODEL);
-    final DeploymentRecordValue deploymentRecord =
+    final DeployedWorkflow workflow =
         testClient
             .receiveFirstDeploymentEvent(DeploymentIntent.CREATED, response.getKey())
-            .getValue();
+            .getValue()
+            .getDeployedWorkflows()
+            .get(0);
 
     // then
     assertThat(RecordingExporter.timerRecords(TimerIntent.CREATED).exists()).isTrue();
     brokerRule.getClock().addTime(Duration.ofSeconds(2));
 
-    assertThat(RecordingExporter.timerRecords(TimerIntent.TRIGGERED).exists()).isTrue();
-    assertThat(RecordingExporter.workflowInstanceRecords(WorkflowInstanceIntent.CREATE).exists())
-        .isTrue();
-
-    final WorkflowInstanceRecordValue wfInstanceRecord =
-        RecordingExporter.workflowInstanceRecords(WorkflowInstanceIntent.ELEMENT_READY)
-            .getFirst()
-            .getValue();
-
-    Assertions.assertThat(wfInstanceRecord)
-        .hasVersion(1)
+    Assertions.assertThat(
+            RecordingExporter.workflowInstanceRecords(WorkflowInstanceIntent.EVENT_TRIGGERING)
+                .getFirst()
+                .getValue())
+        .hasElementId("start_1")
         .hasBpmnProcessId("process")
-        .hasWorkflowKey(deploymentRecord.getDeployedWorkflows().get(0).getWorkflowKey());
+        .hasVersion(workflow.getVersion())
+        .hasWorkflowKey(workflow.getWorkflowKey());
+
+    assertThat(
+            RecordingExporter.getRecords()
+                .stream()
+                .skip(6)
```

Comment:
I still think skip/limit with magic numbers is not really clear, I think we can do better.

What about getting the first `TimerIntent.TRIGGER`, then looking up all records after it? I think that's bit clearer and less fragile.